### ISSUE TITLE
Investigate jquery upgrade

### ISSUE THREAD

**Issue by wallw-bits** [ngageoint/opensphere#90]
Our version of jQuery has two vulnerabilities listed for it:

https://nvd.nist.gov/vuln/detail/CVE-2015-9251 - I'm fairly sure this one is a moot point because we don't use jQuery for requests.

https://nvd.nist.gov/vuln/detail/CVE-2016-10707 - This would cause issues in testing if we had any attributes with the issue. We need to verify that user-generated content cannot produce those attributes.

However, investigate the recommended upgrade to ^3.0.0 and see if our plugins/frameworks are compatible.

**Comment by schmidtk** [comment]
I installed jQuery 3.3.1 and [jquery-migrate](https://github.com/jquery/jquery-migrate), which logs warnings whenever a deprecated/removed jQuery call is made.

These libraries produced a number of deprecation warnings but no errors:

- jquery-ui (1.11.4): New version (1.12.1) available, but it doesn't fix the deprecated code use. Last commit was May 31, 2017 so I would consider this library dead.
- Bootstrap 2: We've discussed this upgrade to great extent, and it's a massive undertaking to switch to Bootstrap 4.
- Slickgrid (2.2): The original project is dead, but there is an actively updated [fork](https://github.com/6pac/SlickGrid/releases) we may be able to migrate to. The [npm package](https://www.npmjs.com/package/slickgrid-6pac) is not regularly updated, so we would still have to maintain the current version in OpenSphere.

These produced errors due to removed interfaces:
- jquery.event.drag: `$event.fixHooks` was removed in 3.0. This is required by Slickgrid, and the new repo mentioned above has a [fixed version](https://6pac.github.io/SlickGrid/lib/jquery.event.drag-2.3.0.js) that works with jQuery 3.x. It also works with our current version of Slickgrid, so this is an easy update.

So far it seems the upgrade is viable, but we may want to lock the version down with `=3.3.1` to prevent problems when deprecated API's are removed in the future.
